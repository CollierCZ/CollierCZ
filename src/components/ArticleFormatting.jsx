import React from "react"
import { css } from '@emotion/core'

class ArticleFormatting extends React.Component {
  render() {
    const { children, className } = this.props;
    return <article 
      className={className}
      css={css`
        position: relative;
        width: 80%;
        max-width: 1200px;
        margin: 0 auto 4rem auto;
        padding-bottom: 0;
        border: none;
        word-wrap: break-word;
        @media only screen and (max-width: 900px) {
          font-size: 0.95em;
        }
        @media only screen and (max-width: 500px) {
          width: auto;
          margin: 2rem 16px;
          padding-bottom: 2rem;
          line-height: 1.65em;
        }
      `}
    >
      {children}
    </article>;
  }
}

export default ArticleFormatting;
