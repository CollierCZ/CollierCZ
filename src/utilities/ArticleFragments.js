import { graphql } from "gatsby"

export const ArticleListFragment = graphql`
  fragment ArticleListFragment on KontentItemArticle {
    fields {
      slug
      tags
      date
    }
    elements {
      metadata__description {
        value
      }
      metadata__keywords {
        value
      }
      categories {
        value {
          name
        }
      }
      title {
        value
      }
      teaser {
        value {
          url
        }
      }            
    }
  }
`