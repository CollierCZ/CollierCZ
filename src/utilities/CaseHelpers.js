const slugify = (text) => {
  return text
    .toString()
    .toLowerCase()
    .trim()
    .replace(/\s+/g, '-')       // Replace spaces with -
    .replace(/[^\w-]+/g, '')     // Remove all non-word chars
}

const titleCase = (text) => {
  return(
    text.replace(/\w\S*/g, function(txt) {
      return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
    })
  )
}

module.exports =
{
  slugify: slugify,
  titleCase: titleCase
}