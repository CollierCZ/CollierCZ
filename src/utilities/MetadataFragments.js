import { graphql } from "gatsby"

export const MetadataFragment = graphql `
  fragment MetadataFragment on KontentItemHome {
    elements {
      title {
        value
      }
      splash_image {
        value {
          url
        }
      }
      blog_logo {
        value {
          url
        }
      }
      metadata__description {
        value
      }
      socialmedia {
        value
      }
      base_url {
        value
      }
    }
  }
`

export const SocialMediaFragment = graphql `
  fragment SocialMediaFragment on KontentItemHome {
    elements {
      socialmedia {
        value
      }
    }
  }
`