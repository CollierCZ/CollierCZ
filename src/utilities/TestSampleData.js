export const author = {
  short_bio: {
    value:
      '<p>Originally from <a href="https://en.wikipedia.org/wiki/Lexington,_Massachusetts" data-new-window="true" target="_blank" rel="noopener noreferrer">Lexington</a> and now residing in <a href="https://www.gotobrno.cz/en/" data-new-window="true" target="_blank" rel="noopener noreferrer">Brno</a>, Aaron is an educator, editor, and elucidator.</p>'
  },
  picture: {
    value: [
      {
        url:
          "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/f7377b84-8ff0-4b77-bc0f-cfdc34d379cc/quixote.jpg"
      }
    ]
  },
  name: {
    value: "Aaron Collier"
  },
  url: {
    value: "aaron-collier"
  }
};

export const simpleImage = {
  description: "A cool picture",
  url: "https://example.com/"
};

export const sampleQuote = {
  system: {
    type: "quote"
  },
  elements: {
    quote: {
      resolvedData: {
        html:
          "<p>You never really understand a person until you consider things from [their] point of view â€” until you climb into [their] skin and walk around in it.</p>"
      }
    },
    source: {
      resolvedData: {
        html:
          "<p>Atticus Finch in <em>To Kill a Mockingbird </em>by Harper Lee</p>"
      }
    }
  }
};

export const sampleShowcase = {
  system: {
    type: "showcase"
  },
  elements: {
    items: {
      linked_items: [
        {
          id: "457d3938-5907-5cca-947d-bae1583e105d",
          elements: {
            link: {
              value: "https://www.youtube.com/watch?v=vj7TyUgBEW8"
            },
            name: {
              value: "Write the Docs talk"
            },
            short_description: {
              links: [
                {
                  linkId: "3cc9d486-53e2-495d-9e2b-25665e0ff29f",
                  urlSlug: "seeing-your-docs-through-different-eyes",
                  type: "article",
                  codename: "write_the_docs_2019"
                }
              ],
              resolvedData: {
                html:
                  '<p>A video of a talk I gave at the <a href="http://www.writethedocs.org/conf/prague/2019/">Write the Docs Prague 2019</a> conference. Or read a <a data-item-id="3cc9d486-53e2-495d-9e2b-25665e0ff29f" href="">text version of the talk</a>.</p>'
              }
            },
            teaser: {
              value: [
                {
                  url:
                    "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/51e9be55-8502-40ca-b374-a1305b36dc56/Slide16.PNG"
                }
              ]
            }
          },
          fields: {
            slug: "write-the-docs-talk"
          }
        },
        {
          id: "fce4fe5b-59c9-5c88-8634-16b63df49d4a",
          elements: {
            link: {
              value: "https://www.youtube.com/watch?v=MlMkAqxoPZM"
            },
            name: {
              value: "Structured content video"
            },
            short_description: {
              links: [],
              resolvedData: {
                html:
                  "<p>A video about what structured content is and how it can benefit content creators.</p>"
              }
            },
            teaser: {
              value: [
                {
                  url:
                    "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/b0481084-18a8-4cf0-ae01-d1a0dbd766ce/structured_content.jpg"
                }
              ]
            }
          },
          fields: {
            slug: "structured-content-video"
          }
        }
      ]
    }
  }
};
