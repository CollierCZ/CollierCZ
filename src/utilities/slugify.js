let slugify = (text) => {
    return text
        .toString()
        .toLowerCase()
        .trim()
        .replace(/\s+/g, '-')           // Replace spaces with -
        .replace(/[^\w-]+/g, '')       // Remove all non-word chars
}


module.exports.slugify = slugify