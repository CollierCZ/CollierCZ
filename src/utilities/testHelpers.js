import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { navigate } from "gatsby";

export const simpleRenderer = component => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
};
export const clickTextLink = component => {
  const mountedComponent = mount(component);
  mountedComponent
    .find("a")
    .first()
    .simulate("click");
  expect(navigate).toHaveBeenCalled();
};
