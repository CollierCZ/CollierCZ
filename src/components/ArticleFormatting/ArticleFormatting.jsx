import React from "react";
import "./ArticleFormatting.css";

class ArticleFormatting extends React.Component {
  render() {
    const { children, className } = this.props;
    return <article className={className}>{children}</article>;
  }
}

export default ArticleFormatting;
