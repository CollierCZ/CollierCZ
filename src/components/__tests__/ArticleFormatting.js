import React from "react"
import { shallow } from "enzyme"
import ArticleFormatting from "../ArticleFormatting"
import { simpleShallowRender } from "../../utilities/testHelpers"

describe("ArticleFormatting", () => {
  it("renders correctly", () => {
    simpleShallowRender(<ArticleFormatting children={{test: "value1"}} />)
  })
})