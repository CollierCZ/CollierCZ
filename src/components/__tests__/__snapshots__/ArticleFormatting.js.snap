// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleFormatting renders correctly 1`] = `
<article
  css={
    Object {
      "map": undefined,
      "name": "1nomv4n",
      "next": undefined,
      "styles": "
      position: relative;
      width: 80%;
      max-width: 1200px;
      margin: 0 auto 4rem auto;
      padding-bottom: 0;
      border: none;
      word-wrap: break-word;
      @media only screen and (max-width: 900px) {
        font-size: 0.95em;
      }
      @media only screen and (max-width: 500px) {
        width: auto;
        margin: 2rem 16px;
        padding-bottom: 2rem;
        line-height: 1.65em;
      }
    ",
      "toString": [Function],
    }
  }
>
  <Component />
</article>
`;
